test/testcafe/spec/ChallengeAuthenticatorPhone_spec.js

```diff
@@ -0,0 +1,255 @@
+import invalidCode from '../../../playground/mocks/data/idp/idx/error-email-verify';
+
+const logger = RequestLogger(/challenge|challenge\/resend|challenge\/answer/,
+  {
+    logRequestBody: true,
+    stringifyRequestBody: true,
+  }
+);
+
+const smsPrimaryMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(phoneVerificationSMSThenVoice)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge')
+  .respond(smsVerification)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/resend')
+  .respond(smsVerification)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+  .respond(success);
+
+const voicePrimaryMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(phoneVerificationVoiceThenSMS)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge')
+  .respond(voiceVerification)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+  .respond(success);
+
+const voiceOnlyMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(phoneVerificationVoiceOnly);
+
+const invalidCodeMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(smsVerification)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+  .respond(invalidCode, 403);
+
+fixture(`Challenge Phone Form`);
+
+async function setup(t) {
+  const challengePhonePageObject = new ChallengePhonePageObject(t);
+  challengePhonePageObject.navigateToPage();
+  return challengePhonePageObject;
+}
+
+test
+  .requestHooks(smsPrimaryMock)(`SMS primary mode - has the right labels`, async t => {
+    const challengePhonePageObject = await setup(t);
+    const pageTitle = challengePhonePageObject.getPageTitle();
+    const pageSubtitle = challengePhonePageObject.getFormSubtitle();
+    const primaryButtonText = challengePhonePageObject.getSaveButtonLabel();
+    const secondaryButtonText = challengePhonePageObject.getSecondaryLinkText();
+    await t.expect(pageTitle).contains('Verify with your phone');
+    await t.expect(pageSubtitle).contains('Send a code via SMS to');
+    await t.expect(pageSubtitle).contains('+1 XXX-XXX-2342');
+    await t.expect(primaryButtonText).contains('Send a code via SMS');
+    await t.expect(secondaryButtonText).contains('Receive a voice call instead');
+  });
+
+test
+  .requestHooks(logger, smsPrimaryMock)(`SMS primary mode - Secondary button click`, async t => {
+    const challengePhonePageObject = await setup(t);
+    await challengePhonePageObject.clickSecondaryLink();
+    await t.expect(logger.count(() => true)).eql(1);
+    const { request: { body, method, url } } = logger.requests[0];
+    const methodTypeRequestBody = JSON.parse(body);
+    await t.expect(methodTypeRequestBody).eql({
+      authenticator: {
+        id: 'autxl8PPhwHUpOpW60g3',
+        methodType: 'voice',
+      },
+      formName: 'authenticator-verification-data',
```

Comment:
hmm, `formName` shall never be in request body. curious why it happens?

Reply:
I don't think it's something to do with my code. Can I open a separate ticket for this?

Reply:
I think I know why. 
It's because you overwrite `local` when creating Model class.
`return ModelClass.extend({ local });`
Look at how you did in the `EnrollAuthenticatorPhoneView`

Reply:
Wow! Good catch! Fixed it..